### ISSUE TITLE
Typings - Lack of optional parameters

### ISSUE THREAD

**Issue by aistrych** [TypeStrong/fork-ts-checker-webpack-plugin#140]
ForkTsCheckerWebpackPlugin require Option type as a parameter but all keys in it are required. For that reason user has to pass all [options](https://github.com/Realytics/fork-ts-checker-webpack-plugin#options) to the plugin which shouldn't be the case. What with default values? It's also impossible to cast passed object to Option type: Options interface is not exported. This is now impossible:

```typescript
new ForkTsCheckerWebpackPlugin({
    tsconfig: "../../tsconfig.json",
    tslint: "../../tslint.json",
    watch: ["./app"],
    workers: ForkTsCheckerWebpackPlugin.TWO_CPUS_FREE
})
```
because Typescript emits:
```
[ts]
Argument of type '{ tsconfig: string; tslint: string; watch: string[]; workers: number; }' is not assignable to parameter of type 'Options'.
  Property 'async' is missing in type '{ tsconfig: string; tslint: string; watch: string[]; workers: number; }'.
```

After adding 'async' it starts to complain about other parameters.

I believe some keys in that interface should be optional:

```typescript
interface Options {
  tsconfig: string;
  tslint: string | true;
  watch: string | string[];
  async: boolean;
  ignoreDiagnostics: number[];
  ignoreLints: string[];
  colors: boolean;
  logger: Console;
  formatter: 'default' | 'codeframe' | Formatter;
  formatterOptions: any;
  silent: boolean;
  checkSyntacticErrors: boolean;
  memoryLimit: number;
  workers: number;
  vue: boolean;
}
```

typescript version: 3.0.1

**Comment by johnnyreilly** [comment]
Sure - if you'd like to submit a PR we can get it in the next release?

**Comment by aistrych** [comment]
Ok, thanks. I will do it tomorrow.

**Comment by Larry1123** [comment]
Oh I just submitted a PR for this #141
